virtualenv.py

```diff
@@ -718,15 +728,15 @@ def call_subprocess(cmd, show_stdout=True,
                     remove_from_env=None, stdin=None):
     cmd_parts = []
     for part in cmd:
-        if len(part) > 45:
-            part = part[:20]+"..."+part[-20:]
-        if ' ' in part or '\n' in part or '"' in part or "'" in part:
-            part = '"%s"' % part.replace('"', '\\"')
         if hasattr(part, 'decode'):
             try:
                 part = part.decode(sys.getdefaultencoding())
             except UnicodeDecodeError:
                 part = part.decode(sys.getfilesystemencoding())
+        if len(part) > 45:
+            part = part[:20]+"..."+part[-20:]
+        if ' ' in part or '\n' in part or '"' in part or "'" in part:
+            part = '"%s"' % part.replace('"', '\\"')
```

Comment:
For example, before my changes, if part is: ```b'/some/very/long/path/to/break/existing/codes/\xe5\x83\x85\xe6\x98\xaf\xe4\xb8\xad\xe6\x96\x87/bin/python'```, ```part[:20]+"..."+part[-20:]``` gives invalid UTF-8 sequences. As a result ellipsis insertion should occur after decoding.